--- src/meson.build
+++ new/meson.build
@@ -564,7 +564,7 @@
   xfixes_dep     = dependency('xfixes', required: false)
   xcomposite_dep = dependency('xcomposite', required: false)
   fontconfig_dep = dependency('fontconfig', fallback: ['fontconfig', 'fontconfig_dep'])
-  atkbridge_dep  = dependency('atk-bridge-2.0', version: at_spi2_atk_req)
+  #atkbridge_dep  = dependency('atk-bridge-2.0', version: at_spi2_atk_req)
 
   backend_immodules += ['xim']
 
