--- src/configure
+++ new/configure
@@ -6785,12 +6785,12 @@
   printf "%s\n" "#define HAVE_UNISTD_H 1" >>confdefs.h
 
 fi
-ac_fn_c_check_header_compile "$LINENO" "libintl.h" "ac_cv_header_libintl_h" "$ac_includes_default"
-if test "x$ac_cv_header_libintl_h" = xyes
-then :
-  printf "%s\n" "#define HAVE_LIBINTL_H 1" >>confdefs.h
-
-fi
+#ac_fn_c_check_header_compile "$LINENO" "libintl.h" "ac_cv_header_libintl_h" "$ac_includes_default"
+#if test "x$ac_cv_header_libintl_h" = xyes
+#then :
+#  printf "%s\n" "#define HAVE_LIBINTL_H 1" >>confdefs.h
+#
+#fi
 
 
 # Checks for typedefs, structures, and compiler characteristics.
