--- e2fsprogs-1.46.5/misc/Makefile.in
+++ patched/misc/Makefile.in
@@ -36,7 +36,7 @@
 
 @FUSE_CMT@FUSE_PROG= fuse2fs
 
-SPROGS=		mke2fs badblocks tune2fs dumpe2fs $(BLKID_PROG) logsave \
+SPROGS=		mke2fs.static badblocks tune2fs.static dumpe2fs.static $(BLKID_PROG).static logsave \
 			$(E2IMAGE_PROG) @FSCK_PROG@ e2undo
 USPROGS=	mklost+found filefrag e2freefrag $(UUIDD_PROG) \
 			$(E4DEFRAG_PROG) $(E4CRYPT_PROG)
@@ -570,16 +570,16 @@
 	$(Q) for i in ext2 ext3 ext4; do \
 		$(ES) "	LINK $(root_sbindir)/mkfs.$$i"; \
 		(cd $(DESTDIR)$(root_sbindir); \
-			$(LN) $(LINK_INSTALL_FLAGS) mke2fs mkfs.$$i); \
+			$(LN) $(LINK_INSTALL_FLAGS) mke2fs.static mkfs.$$i); \
 	done
 	$(Q) (cd $(DESTDIR)$(root_sbindir); \
-		$(LN) $(LINK_INSTALL_FLAGS) dumpe2fs e2mmpstatus)
+		$(LN) $(LINK_INSTALL_FLAGS) dumpe2fs.static e2mmpstatus)
 	$(Q) (cd $(DESTDIR)$(root_sbindir); \
-		$(LN) $(LINK_INSTALL_FLAGS) tune2fs e2label)
+		$(LN) $(LINK_INSTALL_FLAGS) tune2fs.static e2label)
 	$(Q) if test -n "$(FINDFS_LINK)"; then \
 		$(ES) "	LINK $(root_sbindir)/findfs"; \
 		(cd $(DESTDIR)$(root_sbindir); \
-		$(LN) $(LINK_INSTALL_FLAGS) tune2fs $(FINDFS_LINK)); \
+		$(LN) $(LINK_INSTALL_FLAGS) tune2fs.static $(FINDFS_LINK)); \
 	fi
 	$(Q) for i in $(UPROGS); do \
 		$(ES) "	INSTALL $(bindir)/$$i"; \
